src/main/java/net/mcreator/ui/dialogs/SearchUsagesDialog.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.dialogs;
+
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.modgui.ModElementGUI;
+import net.mcreator.util.StringUtils;
+import net.mcreator.util.image.ImageUtils;
+import net.mcreator.workspace.elements.ModElement;
+import net.mcreator.workspace.elements.ModElementManager;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class SearchUsagesDialog {
+
+	/**
+	 * Opens a dialog that shows usages of the elements selected by the user before all across the current workspace.
+	 *
+	 * @param mcreator          Workspace window calling this method.
+	 * @param queryType         Localized string representing type of elements used by mod elements in the given list.
+	 * @param references        List of referencing/dependent mod elements.
+	 * @param deletionRequested Whether user wants to delete selected elements or just view their usages.
+	 * @return Whether elements deletion was requested and the user confirmed deletion.
+	 */
+	public static boolean show(MCreator mcreator, String queryType, List<ModElement> references,

```

Comment:
There should be method variant with deletionRequested to false fixed and void return type, called "showUsages"